script/mix.py

```diff
@@ -1,306 +1,504 @@
-    audio_norm = librosa.util.normalize(audio)
+        audio, sr = librosa.load(wav_file, sr=sr)
+        audio_norm = librosa.util.normalize(audio)
 
-    # normalizing audio
+        new_rate = float(target_tempo / current_tempo)
+        stretched_audio = librosa.effects.time_stretch(audio_norm, rate=new_rate)
 
-    stretched_audios.append(audio_norm)
+        stretched_audios.append(stretched_audio)
 
     if len(stretched_audios) != n_stems:
         invalid_mixture = True
+        print("invalid, we lost a stretched audio")
 
     return stretched_audios, invalid_mixture
 
+def shift(stretched_audios):
+
+    """
+
+    Receives audios post-stretching and collects information on their first downbeats and stores as
+    a list. It then finds the index of the audio of the latest downbeat and holds that audio
+    constant while shifting every other audio by the difference between their first downbeat
+    and the latest-occurring first downbeat by padding audiofile with zeroes. It then double checks
+    first-downbeats to make sure that they are all the same. 
+
+    Parameters
+    ----------
+
+    stretched_audios(list): list of stretched audios
+
+    Returns
+    -------
+    final_audios(list): list of shifted audios
+    invalid_mixture(bool): check of mixture validity
+
+    """
+
 
-def shift(sr, stretched_audios, invalid_mixture):
     first_downbeats = []
     final_audios = []
+    invalid_mixture = False
 
     for audio in stretched_audios:
-        _, beat_times = librosa.beat.beat_track(y=audio, sr=sr)
-        downbeat_times = librosa.frames_to_time(beat_times, sr=sr)
+        _, beat_times = librosa.beat.beat_track(y=audio, sr=DEFAULT_SR)
+        downbeat_times = librosa.frames_to_time(beat_times, sr=DEFAULT_SR)
         first_downbeats.append(downbeat_times[0])
 
     latest_beat = max(first_downbeats)
-
     latest_beat_index = first_downbeats.index(latest_beat)
 
-    immutable_audio = stretched_audios[latest_beat_index]
+    fixed_audio = stretched_audios[latest_beat_index]
 
-    final_audios.append(immutable_audio)
+    final_audios.append(fixed_audio)
 
     for i in range(0, len(stretched_audios)):
         if i != latest_beat_index:
             shift_difference = np.abs(first_downbeats[i] - latest_beat)
-            silence_samples = int(shift_difference * sr)
+            silence_samples = int(shift_difference * DEFAULT_SR)
             silence = np.zeros(silence_samples)
 
             final_audio = np.concatenate([silence, stretched_audios[i]])
 
+
             # rechecking downbeats after shift
             print("rechecking downbeat alignment")
-            _, beat_times = librosa.beat.beat_track(y=final_audio, sr=sr)
-            downbeat_times = librosa.frames_to_time(beat_times, sr=sr)
+            _, beat_times = librosa.beat.beat_track(y=final_audio, sr=DEFAULT_SR)
+            downbeat_times = librosa.frames_to_time(beat_times, sr=DEFAULT_SR)
             print(downbeat_times[0])
```

Comment:
you're returning `final_audios` regardless of this piece of code. we can remove it.